### ISSUE TITLE
false negative `reportGeneralTypeIssues` when `override` on module function

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#751]
```py
from typing import override


class A:
    @override
    def f() -> None: ... # Method "f" is marked as override, but no base method of same name is present  (reportGeneralTypeIssues)

@override
def f() -> None: ...  # no error
```
